boa_tester/src/edition.rs

```diff
@@ -0,0 +1,332 @@
+    "Proxy" => SpecEdition::ES6,
+    "proxy-missing-checks" => SpecEdition::ES6,
+    "Reflect" => SpecEdition::ES6,
+    "Reflect.construct" => SpecEdition::ES6,
+    "Reflect.set" => SpecEdition::ES6,
+    "Reflect.setPrototypeOf" => SpecEdition::ES6,
+    "regexp-dotall" => SpecEdition::ES13,
+    "regexp-lookbehind" => SpecEdition::ES13,
+    "regexp-match-indices" => SpecEdition::ES13,
+    "regexp-named-groups" => SpecEdition::ES13,
+    "regexp-unicode-property-escapes" => SpecEdition::ES13,
+    "rest-parameters" => SpecEdition::ES6,
+    "Set" => SpecEdition::ES6,
+    "SharedArrayBuffer" => SpecEdition::ES13,
+    "string-trimming" => SpecEdition::ES5,
+    "String.fromCodePoint" => SpecEdition::ES6,
+    "String.prototype.at" => SpecEdition::ES13,
+    "String.prototype.endsWith" => SpecEdition::ES6,
+    "String.prototype.includes" => SpecEdition::ES6,
+    "String.prototype.matchAll" => SpecEdition::ES13,
+    "String.prototype.replaceAll" => SpecEdition::ES13,
+    "String.prototype.trimEnd" => SpecEdition::ES13,
+    "String.prototype.trimStart" => SpecEdition::ES13,
+    "super" => SpecEdition::ES6,
+    "Symbol" => SpecEdition::ES6,
+    "Symbol.asyncIterator" => SpecEdition::ES13,
+    "Symbol.hasInstance" => SpecEdition::ES6,
+    "Symbol.isConcatSpreadable" => SpecEdition::ES6,
+    "Symbol.iterator" => SpecEdition::ES6,
+    "Symbol.match" => SpecEdition::ES6,
+    "Symbol.matchAll" => SpecEdition::ES13,
+    "Symbol.prototype.description" => SpecEdition::ES13,
+    "Symbol.replace" => SpecEdition::ES6,
+    "Symbol.search" => SpecEdition::ES6,
+    "Symbol.species" => SpecEdition::ES6,
+    "Symbol.split" => SpecEdition::ES6,
+    "Symbol.toPrimitive" => SpecEdition::ES6,
+    "Symbol.toStringTag" => SpecEdition::ES6,
+    "Symbol.unscopables" => SpecEdition::ES6,
+    "tail-call-optimization" => SpecEdition::ES6,
+    "template" => SpecEdition::ES6,
+    "top-level-await" => SpecEdition::ES13,
+    "TypedArray" => SpecEdition::ES6,
+    "TypedArray.prototype.at" => SpecEdition::ES13,
+    "u180e" => SpecEdition::ES5,
+    "Uint8Array" => SpecEdition::ES6,
+    "Uint16Array" => SpecEdition::ES6,
+    "Uint32Array" => SpecEdition::ES6,
+    "Uint8ClampedArray" => SpecEdition::ES6,
+    "WeakMap" => SpecEdition::ES6,
+    "WeakRef" => SpecEdition::ES13,
+    "WeakSet" => SpecEdition::ES6,
+    "well-formed-json-stringify" => SpecEdition::ES5,
+    "__proto__" => SpecEdition::ES6,
+    "__getter__" => SpecEdition::ES13,
+    "__setter__" => SpecEdition::ES13,
+
+    // Test-Harness Features
+
+    "IsHTMLDDA" => SpecEdition::ES13,
+    "host-gc-required" => SpecEdition::ES13,
+};
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Default,
+    Serialize,
+    Deserialize,
+    clap::ValueEnum,
+)]
+#[serde(untagged)]
+pub(crate) enum SpecEdition {
+    ES5 = 5,
+    ES6,
+    ES7,
+    ES8,
+    ES9,
+    ES10,
+    ES11,
+    ES12,
+    #[default]
+    ES13,
```

Comment:
I would maybe add "ESNext" here, since we track the next version of the spec, and some of the features are not yet available in the yearly spec.